### DISCUSSION TITLE
Combine build results in one Docker image

### DISCUSSION THREAD

**Discussion by till** [goreleaser/goreleaser#3339] [Q&A] [answered]
I have a project which has a server and a cli helper tool (for the server). They are individual tools in the same repository and each is setup in goreleaser as a build.

Essentially I would like to be able to combine both after build and copy both executables into one (Docker) image.

I couldn't find any issues describing this, hence the question.

A workaround is to build individual images and to combine them outside goreleaser. So it's not absolutely necessary (I guess 😅) still figured I asked in case I missed it or anyone thinks this is a good idea.

**Comment by caarlos0** [comment, marked as answer]
if you dont add any filters to your `dockers` section, all artifacts for the given target (os/arch/etc) will be available... so this is basically the default behavior :)

**Reply by till** [reply]
Weird, I tried that. Sorry 🙈 for the noise. I'll try again. Thanks for responding so quickly!

**Reply by waldirborbajr** [reply]
@till Did you fix it? Iam with same problem, I have a CLI and API in the same project, but I'm frying my head how to add both into my Dockerfile. If let only one works, but if I add another binary I got an error that my binary not found.

**Reply by till** [reply]
Just add it, default behavior is that all are available. Maybe my version was outdated. Didn't do anything else to fix it.

**Reply by waldirborbajr** [reply]
@till Can I take a look at your yaml config?

**Reply by caarlos0** [reply]
@waldirborbajr by default all the binaries built for the current platform will be available, so you can just `ADD` or `COPY` the binary name to `/bin` or something like that.

You can also install your nfpm packages, e.g.: https://github.com/goreleaser/goreleaser/blob/main/Dockerfile#L24

**Reply by waldirborbajr** [reply]
Hmm... so I think my problem must be during the build process because 1 binary works perfectly and the second during copy I face with file not found. Back to start again... 
Thank you for now.

**Reply by till** [reply]
@waldirborbajr I had made an example and just cleaned it up, for you or anyone else:

This is the result:

```
❯ docker run --rm -it till/goreleaser-multi-bin-docker
root@8c2794ad092c:/# ls -lah /opt/bin/
total 2.4M
drwxr-xr-x 1 root root 4.0K Oct 29 16:03 .
drwxr-xr-x 1 root root 4.0K Oct 29 16:03 ..
-rwxr-xr-x 1 root root 1.2M Oct 29 16:03 bar
-rwxr-xr-x 1 root root 1.2M Oct 29 16:03 foo
root@8c2794ad092c:/# /opt/bin/bar 
Hello from bar
root@8c2794ad092c:/# /opt/bin/foo 
Hello from foo
root@8c2794ad092c:/# 
```

I used this command to build it: `goreleaser r --snapshot --rm-dist`

The complete source code and config is here:
https://github.com/till/goreleaser-multi-bin-docker

**Reply by waldirborbajr** [reply]
Wow...  guys good news, I think that must be working now, no more errors during copy files to docker. 

Now I have to:
1. verify why the action does not deploy to GHCR
2. Fix some file names.

But the main bug was solved. yeah.

@caarlos0 @till 
Thank you guys for your attention

**Reply by caarlos0** [reply]
what was causing the issue?

**Reply by waldirborbajr** [reply]
@caarlos0 Probably the missing parameter --snapshot I'll try it again removing the parameter and build it again. As soon as I finish my tests I'll let you know.
